libkmod/libkmod-elf.c

```diff
@@ -1101,8 +1116,20 @@ int kmod_elf_get_dependency_symbols(const struct kmod_elf *elf,
 		for (i = 0; i < vercount; i++) {
 			if (visited_versions[i] == 0) {
 				const char *name;
+				size_t nlen;
+
 				name = elf_get_mem(elf, ver_off + i * verlen + crclen);
-				slen += strlen(name) + 1;
+				nlen = strnlen(name, namlen);
+
+				if (nlen == namlen) {
+					ELFDBG(elf, "symbol name at index %d too long\n",
+					       i);
+					free(visited_versions);
+					free(symcrcs);
+					*array = NULL;
```

Comment:
Nit: no need to clear output variable on error.

Reply:
Removed